### ISSUE TITLE
Show status of current-site toggle in the icon

### ISSUE THREAD

**Issue by jtrakk** [darkreader/darkreader#1671]
This request is an extension of https://github.com/darkreader/darkreader/issues/609. It is nice that the current-site toggle value is visible when I click on the icon, but I would like for it to be visible without having to click on the icon. 

Perhaps the goggles could change color or something.

**Comment by Gusted** [comment]
#3218

**Comment by Servinjesus1** [comment]
So, @Gusted, do I understand the history of this FR correctly that it remains open as a feature request you would consider if there was a way to provide it without requiring the [new "invasive" permission](https://github.com/darkreader/darkreader/pull/3218) you mention? It's a shame if such is true, especially if something as widely trusted as uBlock Origin can provide the feature, though I recognize I don't have the merit to share nor hold a strong opinion on either the security of the feature in uBlock or its noninvasive adaptability to Dark Reader.

**Comment by Gusted** [comment]
> So, @Gusted, do I understand the history of this FR correctly that it remains open as a feature request you would consider if there was a way to provide it without requiring the [new "invasive" permission](https://github.com/darkreader/darkreader/pull/3218) you mention?

That's technically not possible. The API's are quite strict, there's no "oh wow by using this API we can abuse it and actually still have the same functionally", in my experience with the webextension API. Such thing won't happen. 



> It's a shame if such is true, especially if something as widely trusted as uBlock Origin can provide the feature, though I recognize I don't have the merit to share nor hold a strong opinion on either the security of the feature in uBlock or its noninvasive adaptability to Dark Reader.

The problem here is, requiring **new** permissions causes a lot of issues with browser and the behavior is quite undefined. IIRC certain version of Chromium would actually just disable the extension and required to explicitly be enabled again by the user. However since then we already somehow miraculously are asking the permission(which would be required for the MV3 migration). So it should only be a matter of time to get a PR up to provide this functionality.

**Comment by yet-another-tim** [comment]
+1

Usecase that makes me wish it existed:
When visiting a site with a dark theme, I'm not aware if the dark theme extension is toggled on, or whether it's their dark theme. Usually when the site owner has gone to the effort to use a dark theme it looks better natively, with the extension toggled off.

It'd be nice to glance at the icon and know immediately if I'm looking at site dark theme or extension dark theme.

**Comment by jibbers42** [comment]
I agree with this [comment](https://github.com/darkreader/darkreader/issues/1671#issuecomment-1492764790) and I'm often checking the popup menu to see if dark reader is enabled for the site or not.

I was thinking about how to indicate this and using colors or black/white would be confusing given the nature of the extension. If my green-headed, red-eyed icon was black and white half the time, I would probably have to think about if that means it's disabled or if it's actively making the site dark.

My thought is to change the icon's circle goggle eyes to two x's if the extension is disabled for a site.

**Comment by Cyber5imon** [comment]
This request has been ongoing since July 7, 2018 (2331 days). A lot has changed since then. In 2024, a lot of popular browser extensions are modifying the toolbar icon. Users have become accepting of allowing the permissions required for these extensions to provide the functionality of changing the toolbar icon.

Some examples are:
- Bitwarden: Shows the number of matching logins for the website.
- Distill Web Monitoring: Shows the number of unread monitors.
- uBlock Origin: Shows if it is enabled/disabled for the site (exactly what we are looking for with this request).
- Tampermonkey: Shows number of matching scripts for website.

There is enough interest in this function:
- https://github.com/darkreader/darkreader/issues/609
- https://github.com/darkreader/darkreader/issues/12868
- https://github.com/darkreader/darkreader/issues/12542
- https://github.com/darkreader/darkreader/issues/13305

I hope this doesn't fall on deaf ears and the functionality to show the enabled/disabled state of Dark Reader for the current website can be added to the next release. You could probably even start a function bounty for this and drum up some donations specifically to add this feature!

**Comment by alexanderby** [comment]
This feature will be available in the next release.

**Comment by Gitoffthelawn** [comment]
@alexanderby Over the years, there have been quite a few feature requests on this topic, and although I could be mistaken, I think what many people were requesting has not yet been implemented: having the Dark Reader icon indicate whether or not Dark Reader is affecting the colours of ***the current site***.

Here are a couple helpful links for reference:  
https://github.com/darkreader/darkreader/issues/1671#issuecomment-1492764790
https://github.com/darkreader/darkreader/pull/3218#issuecomment-687609318

What do you think?

P.S. Personally, I would like the Dark Reader icon to indicate whether Dark Reader is globally enabled or disabled and, if enabled, which mode it is applying to the current site: dynamic, static, filter (or filter+), or none.
